### PR TITLE
Created test for `Gen.intersperse`

### PR DESCRIPTION
**Pull request by paplorinc**
Fixes https://github.com/javaslang/javaslang/issues/1275 and increases code coverage.

### CODE REVIEW

File: javaslang-test/src/main/java/javaslang/test/Gen.java [hunk 1]
```diff
@@ -298,20 +299,16 @@
    * @return A new T generator
    */
     default Gen<T> intersperse(Gen<T> other) {
-        java.util.Iterator<Gen<T>> iter = new java.util.Iterator<Gen<T>>() {
+        Supplier<Gen<T>> iter = new Supplier<Gen<T>>() {
             boolean genSwitch = false;
 
             @Override
-            public boolean hasNext() {
```
Comment (paplorinc) [inline javaslang-test/src/main/java/javaslang/test/Gen.java]: `next` was't covered by tests

File: javaslang-test/src/main/java/javaslang/test/Gen.java [hunk 2]
```diff
@@ -298,20 +299,16 @@
...
-                return true;
-            }
-
-            @Override
-            public Gen<T> next() {
-                return (genSwitch = !genSwitch) ? Gen.this : other;
```
Comment (paplorinc) [inline javaslang-test/src/main/java/javaslang/test/Gen.java]: asking the question changed the answer...

File: javaslang-test/src/main/java/javaslang/test/Gen.java [hunk 3]
```diff
@@ -298,20 +299,16 @@
    * @return A new T generator
    */
     default Gen<T> intersperse(Gen<T> other) {
-        java.util.Iterator<Gen<T>> iter = new java.util.Iterator<Gen<T>>() {
```
Comment (paplorinc) [inline javaslang-test/src/main/java/javaslang/test/Gen.java]: iterator was basically used as a supplier

Comment (danieldietrich) [inline javaslang-test/src/main/java/javaslang/test/Gen.java]: Nice!

**Comment by danieldietrich** [comment]
Thanks!

**Pull request merged**